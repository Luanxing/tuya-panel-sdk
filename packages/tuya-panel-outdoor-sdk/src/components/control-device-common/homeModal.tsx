import React, { ReactElement } from 'react';
import { Utils, Modal } from 'tuya-panel-kit';
import { View, StyleSheet, ImageBackground, TouchableOpacity, Text } from 'react-native';
import { IIcon, IModalProps } from './common/interface';

const { convertX: cx } = Utils.RatioUtils;

const homeModal = (props: IModalProps) => {
  const result = (Child: (childProps: IIcon) => ReactElement) => {
    const {
      title,
      subTitle,
      done,
      bgImage,
      bgStyle,
      bgChildStyle,
      showModal,
      v1Bottom,
      v2Bottom,
      iconBoxStyle,
      iconProp,
      icon,
      onMaskPress,
    } = props;
    const curIcon = { ...iconProp, ...{ icon } };
    return (
      <Modal style={styles.modalView} visible={showModal} onMaskPress={onMaskPress}>
        <View style={[iconBoxStyle, { bottom: v1Bottom }, iconProp.iconStyle]}>
          <Child {...curIcon} />
        </View>
        <View style={[styles.alertWrap, { bottom: v2Bottom }]}>
          <ImageBackground style={bgStyle} source={bgImage}>
            <View style={[{ width: '100%', height: '100%' }, bgChildStyle]}>
              <View style={styles.titleWrap}>
                <Text style={styles.titleViewName}>{title}</Text>
              </View>
              <Text numberOfLines={2} style={styles.titleTip}>
                {subTitle}
              </Text>
              <TouchableOpacity style={styles.modalButton} onPress={onMaskPress}>
                <Text style={styles.modalText}>{done}</Text>
              </TouchableOpacity>
            </View>
          </ImageBackground>
        </View>
      </Modal>
    );
  };
  return result;
};

export default homeModal;

const styles = StyleSheet.create({
  alertWrap: {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'center',
  },
  modalButton: {
    height: cx(20),
    justifyContent: 'center',
    marginBottom: cx(24),
    marginTop: cx(24),
  },
  modalText: {
    color: '#0376FF',
    fontSize: cx(16),
    fontWeight: '500',
  },
  modalView: {
    position: 'absolute',
  },
  titleTip: {
    color: 'rgba(0,0,0,0.5)',
    flex: 1,
    fontSize: cx(14),
    marginTop: cx(8),
  },
  titleViewName: {
    color: 'rgba(0,0,0,0.9)',
    fontSize: cx(16),
    fontWeight: '500',
  },
  titleWrap: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    lineHeight: cx(20),
  },
});
