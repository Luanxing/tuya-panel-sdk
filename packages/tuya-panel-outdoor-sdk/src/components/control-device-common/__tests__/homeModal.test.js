/**
 * @jest-environment jsdom
 */
import React from 'react';
import { mount, shallow } from 'enzyme';
import { TYSdk, Utils } from 'tuya-panel-kit';
import Child from '../child';
import homeModal from '../homeModal';

const { convertX: cx } = Utils.RatioUtils;

describe('homeModal components', () => {
  beforeEach(() => {});
  it('homeModal render', async () => {
    const onMaskPress = () => {
      //
    };

    const wrapper = mount(
      homeModal({
        title: 'title',
        subTitle: 'subTitle',
        done: 'done',
        bgImage: 'bgImage',
        bgStyle: { width: cx(280), height: cx(170), justifyContent: 'center' },
        bgChildStyle: {
          width: cx(280),
          height: cx(150),
          marginTop: cx(40),
        },
        showModal: true,
        v1Bottom: cx(300),
        v2Bottom: cx(100),
        iconBoxStyle: { position: 'absolute', right: cx(10) },
        iconProp: {
          color: '#333',
          size: cx(24),
          iconStyle: {
            width: cx(50),
            height: cx(50),
            borderRadius: cx(50),
            alignItems: 'center',
            justifyContent: 'center',
          },
        },
        icon: '',
        onMaskPress: () => onMaskPress(1),
      })(Child)
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
