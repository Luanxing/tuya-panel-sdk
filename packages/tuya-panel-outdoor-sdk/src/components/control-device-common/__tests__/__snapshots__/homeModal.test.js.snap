// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homeModal components homeModal render 1`] = `
<Portal
  onMaskPress={[Function]}
  style={
    Object {
      "position": "absolute",
    }
  }
  visible={true}
/>
`;
